build.py

```diff
@@ -1,18 +1,68 @@
-import math
-import subprocess
 from pathlib import Path
 
+import argparse
+
 import fontmake.instantiator
 import fontTools.designspaceLib
 import ufo2ft
 import ufoLib2
 import vttLib
-import copy
 
 INPUT_DIR = Path("sources")
 OUTPUT_DIR = Path("build")
+VTT_DATA_FILE = (INPUT_DIR / "vtt_data" / "CascadiaCode.ttx")
+
+def step_set_font_name(n):
+    def _set(instance):
+        instance.info.familyName = n
+    return _set
+
+def step_merge_glyphs_from_ufo(path):
+    def _merge(instance):
+        ufo = ufoLib2.Font.open(path)
+        print (f"[{instance.info.familyName}] Merging {path}")
+        for glyph in ufo.glyphOrder:
+            if glyph not in instance.glyphOrder:
+                instance.addGlyph(ufo.get(glyph))
+
+    return _merge
+
+def step_set_feature_file(n):
+    fea = n.read_text()
+    def _set(instance):
+        instance.features.text = fea
+    return _set
+
+def build_font_instance(generator, instance_descriptor, *steps):
+    instance = generator.generate_instance(instance_descriptor)
+
+    for step in steps:
+        step(instance)
+
+    setattr(instance.info, "openTypeOS2Panose", [2,11,6,9,2,0,0,2,0,4])
+
+    familyName = instance.info.familyName
+    file_name = f"{familyName}.ttf".replace(" ", "")
+    file_path = OUTPUT_DIR / file_name
+
+    print (f"[{familyName}] Compiling")
+    instance_font = ufo2ft.compileTTF(instance, removeOverlaps=True, inplace=True)
+
+    print (f"[{familyName}] Merging VTT")
+    vttLib.transfer.merge_from_file(instance_font, VTT_DATA_FILE)
+
+    print (f"[{familyName}] Saving")
+    instance_font.save(file_path)
+
+    print (f"[{familyName}] Done: {file_path}")
 
 if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='build some fonts')
+    parser.add_argument('-N', '--no-nerdfonts', default = False, action = 'store_true')
```

Comment:
`store_true` implies `default = False`

Reply:
appreciated. I might leave this one because the args are going to evolve in the right direction at some point.